### PR TITLE
increase coverage

### PR DESCRIPTION
**Pull request by mgred**
#### Purpose (TL;DR) - mandatory

Increase Coverage towards 💯 %

#### How to verify - mandatory
1. Check out this branch
2. `npm install`
3. `npm run test-coverage`

#### Checklist for author

- [x] `npm run lint` passes
- [x] References to standard library functions are [cached](https://github.com/sinonjs/sinon/pull/1523).

### CODE REVIEW

File: lib/create-matcher.js [hunk 1]
```diff
@@ -108,8 +108,8 @@ createMatcher.typeOf = function(type) {
 
 createMatcher.instanceOf = function(type) {
     if (
-        typeof Symbol === "undefined" ||
-        typeof Symbol.hasInstance === "undefined"
+        typeOf(Symbol) === "undefined" ||
```
Comment (mantoni) [inline lib/create-matcher.js]: Isn't this causing `ReferenceError: Symbol is not defined` on platforms that don't support Symbol? I think the `typeof Symbol` check is needed for exactly that case, no?

Comment (mgred) [reply lib/create-matcher.js]: grrr, good question 😄 Not sure on this one honestly

**Comment by mroderick** [comment]
```
1) matcher
       .instanceOf
         throws if given argument is not a function:
     AssertionError: [assert.exception] Expected { name: "TypeError" } but threw ReferenceError ('Symbol' is undefined)
ReferenceError: 'Symbol' is undefined
      at createMatcher.instanceOf (lib/create-matcher.js:110)
      at Anonymous function (lib/create-matcher.test.js:583)
      at captureException (node_modules/@sinonjs/referee/lib/capture-exception.js:5)
      at assert (node_modules/@sinonjs/referee/lib/assertions/exception.js:32)
      at assertion (node_modules/@sinonjs/referee/lib/define-assertion.js:51)
      at referee[type][name] (node_modules/@sinonjs/referee/lib/define-assertion.js:93)
      at Anonymous function (lib/create-matcher.test.js:581)
   at Runner.prototype.runTest (Unknown script code:34401:5)
      at fail (node_modules/@sinonjs/referee/lib/create-fail.js:15)
      at ctx.fail (node_modules/@sinonjs/referee/lib/define-assertion.js:46)
      at assert (node_modules/@sinonjs/referee/lib/assertions/exception.js:54)
      at assertion (node_modules/@sinonjs/referee/lib/define-assertion.js:51)
      at referee[type][name] (node_modules/@sinonjs/referee/lib/define-assertion.js:93)
      at Anonymous function (lib/create-matcher.test.js:581)

  2) matcher
       .instanceOf
         returns matcher:
     ReferenceError: 'Symbol' is undefined
      at createMatcher.instanceOf (lib/create-matcher.js:110)
      at Anonymous function (lib/create-matcher.test.js:609)

  3) matcher
       .instanceOf
         returns true if test is called with instance of argument:
     ReferenceError: 'Symbol' is undefined
      at createMatcher.instanceOf (lib/create-matcher.js:110)
      at Anonymous function (lib/create-matcher.test.js:615)

  4) matcher
       .instanceOf
         returns false if test is not called with instance of argument:
     ReferenceError: 'Symbol' is undefined
      at createMatcher.instanceOf (lib/create-matcher.js:110)
      at Anonymous function (lib/create-matcher.test.js:621)

  5) matcher
       .instanceOf
         when Symbol is not defined
           should call assertType and return a matcher:
     ReferenceError: 'Symbol' is undefined
      at createMatcher.instanceOf (lib/create-matcher.js:110)
      at Anonymous function (lib/create-matcher.test.js:638)

  6) matcher
       .or
         requires matcher argument:
     AssertionError: [assert.exception] Expected { name: "TypeError" } but threw ReferenceError ('Symbol' is undefined)
ReferenceError: 'Symbol' is undefined
      at createMatcher.instanceOf (lib/create-matcher.js:110)
      at Anonymous function (lib/create-matcher.test.js:1425)
      at captureException (node_modules/@sinonjs/referee/lib/capture-exception.js:5)
      at assert (node_modules/@sinonjs/referee/lib/assertions/exception.js:32)
      at assertion (node_modules/@sinonjs/referee/lib/define-assertion.js:51)
      at referee[type][name] (node_modules/@sinonjs/referee/lib/define-assertion.js:93)
      at Anonymous function (lib/create-matcher.test.js:1423)
   at Runner.prototype.runTest (Unknown script code:34401:5)
      at fail (node_modules/@sinonjs/referee/lib/create-fail.js:15)
      at ctx.fail (node_modules/@sinonjs/referee/lib/define-assertion.js:46)
      at assert (node_modules/@sinonjs/referee/lib/assertions/exception.js:54)
      at assertion (node_modules/@sinonjs/referee/lib/define-assertion.js:51)
      at referee[type][name] (node_modules/@sinonjs/referee/lib/define-assertion.js:93)
      at Anonymous function (lib/create-matcher.test.js:1423)

  7) matcher
       .and
         requires matcher argument:
     AssertionError: [assert.exception] Expected { name: "TypeError" } but threw ReferenceError ('Symbol' is undefined)
ReferenceError: 'Symbol' is undefined
      at createMatcher.instanceOf (lib/create-matcher.js:110)
      at Anonymous function (lib/create-matcher.test.js:1474)
      at captureException (node_modules/@sinonjs/referee/lib/capture-exception.js:5)
      at assert (node_modules/@sinonjs/referee/lib/assertions/exception.js:32)
      at assertion (node_modules/@sinonjs/referee/lib/define-assertion.js:51)
      at referee[type][name] (node_modules/@sinonjs/referee/lib/define-assertion.js:93)
      at Anonymous function (lib/create-matcher.test.js:1472)
   at Runner.prototype.runTest (Unknown script code:34401:5)
      at fail (node_modules/@sinonjs/referee/lib/create-fail.js:15)
      at ctx.fail (node_modules/@sinonjs/referee/lib/define-assertion.js:46)
      at assert (node_modules/@sinonjs/referee/lib/assertions/exception.js:54)
      at assertion (node_modules/@sinonjs/referee/lib/define-assertion.js:51)
      at referee[type][name] (node_modules/@sinonjs/referee/lib/define-assertion.js:93)
      at Anonymous function (lib/create-matcher.test.js:1472)
```

File: lib/deep-equal.test.js [hunk 2]
```diff
@@ -800,4 +809,14 @@ describe("deepEqual", function() {
             assert.isFalse(checkDeep);
         });
     });
+
+    describe("when called with matchers", function() {
```
Comment (khamiltonuk) [inline lib/deep-equal.test.js]: Double test

File: lib/is-arguments.test.js [hunk 3]
```diff
@@ -30,6 +30,12 @@ describe("isArguments", function() {
                 assert.isFalse(checkArgs);
             });
 
+            it("returns true if callee is a function", function() {
```
Comment (khamiltonuk) [inline lib/is-arguments.test.js]: Can we remove this test and use the technique from the open pull request #119

File: lib/create-matcher.js (hunk 1 above)
Comment (mgred) [reply lib/create-matcher.js]: Seems you're right https://github.com/sinonjs/samsam/pull/118#issuecomment-538664573

File: lib/deep-equal.test.js (hunk 2 above)
Comment (mgred) [reply lib/deep-equal.test.js]: I will remove this in favor of your test 👍

**Comment by mroderick** [comment]
Whoop!

**Pull request merged**